vnext/Shared/OInstance.cpp

```diff
@@ -385,6 +389,24 @@ InstanceImpl::InstanceImpl(
           m_devSettings->jsiRuntimeHolder =
               std::make_shared<Microsoft::JSI::ChakraRuntimeHolder>(m_devSettings, m_jsThread, nullptr, nullptr);
           break;
+        case JSIEngineOverride::V8Napi: {
+#if defined(USE_V8)
+          std::unique_ptr<facebook::jsi::PreparedScriptStore> preparedScriptStore = nullptr;
+
+          char tempPath[MAX_PATH];
+          if (GetTempPathA(MAX_PATH, tempPath)) {
```

Comment:
don't use non-Unicode functions

Reply:
Consumed V8 APIs expect 8-bit chars.
Can you please elaborate?

Also, We should punt this to a subsequent change, for this part of the code is identical to the existing V8 case.

Reply:
this doesn't return utf8 chars, this returns ascii (only 0-255). The other code is wrong too AFAICT.

Reply:
Keeping code as originally submitted for plain `V8`.
Follow up in #8457.

Reply:
Iirc @NikoAri raised the issue a few months ago, mentioning that it would break bytecode caching for users of different locales.

The fix here is fairly trivial, so I agree we should just go the right thing here.

Reply:
Yes, please don't use GetTempPathA, it will break for non-Ascii Utf-8 characters, use Unicode GetTempPathW

Reply:
> Yes, please don't use GetTempPathA, it will break for non-Ascii Utf-8 characters, use Unicode GetTempPathW

We can't do that. `GetTempPathW` will yield a 16-bit string, which is not accepted by the `ScriptStore` implementation.
Filed #8457 for follow-up.

Reply:
You are adding a new usage of the wrong API, please fix.

Reply:
@asklar as already pointed, I'm not adding this usage. It was already there in the `V8` case.

I'm happy to replace this with a UTF-8-compatbile API, but `GetTempPathW` is not it.
Do we have an alternative function?

Reply:
Windows "W" APIs are UTF-16. So you'd have to call the UTF-16 API, then convert to UTF8 or whichever encoding your API takes.

Reply:
> Windows "W" APIs are UTF-16. So you'd have to call the UTF-16 API, then convert to UTF8 or whichever encoding your API takes.

Noted. Then there is no equivalent UTF8-friendly function as I feared.
I'll apply the suggested workaround, similarly to other parts of the code base (`winrt::to_string`).

Reply:
@asklar `GetTempPathA` has been dropped. Ready for approval.